ykcompile/src/lib.rs

```diff
@@ -579,7 +579,7 @@ impl<TT> TraceCompiler<TT> {
             Location::Mem(..) => todo!(),
             Location::Const { val, .. } => {
                 let val = val.i64_cast();
-                if val as u32 > u32::MAX {
+                if val > u32::MAX as i64 {
```

Comment:
You could also use

```suggestion
                if val > i64::from(u32::MAX) {
```

Reply:
Is there any advantage to doing it that way?

Reply:
Clippy warns against `as` as it is easy to accidentally write a truncating cast.

Reply:
Thanks. Fixed in ac540b6

Reply:
I agree with Clippy and Bjorn: `as` is not a good feature and we should avoid its use whenever possible.